Linear search array

cmchan04 opened the pull request:
Just accidentally merged into main when resolving conflicts, (misclicked merge when pulling QQ) Here is the new PR for linear search @BengRhui

cmchan04 committed:
Merge pull request #16 from cmchan04/main

Merge main to linear search for array

cmchan04 committed:
Update comment

cmchan04 committed:
Update comment

BengRhui committed:
Replace ranges::transform with toLowerCase

The parts where ranges::transform is used are all changed to toLowerCase() method.

BengRhui commented:
@cmchan04 I've updated `ranges::transform` to `toLowerCase`, can u pls take a look on your side and see if everything is able to compile nicely?

BengRhui committed:
Modified the result storing process

A little more comments are added, with the result storing process keeping consistent to others for fair comparison.

BengRhui reviewed (approved):
@cmchan04 I've slightly modified the code to make sure the process is similar across different searches. If everything looks good on your side, do feel free to merge into the main branch.

cmchan04 reviewed (commented):
Do we need to convert ```transactionType``` and ```location``` to lower when we loading the data? then we no need ```toLowerCase``` too often

cmchan04 reviewed (commented):
I don't think array need a ptr to ptr right? Array usually use single pointer only, because there is no Node like structure mah

cmchan04 commented:
> @cmchan04 I've slightly modified the code to make sure the process is similar across different searches. If everything looks good on your side, do feel free to merge into the main branch.

The ptr to ptr also seems not utilized, just like adding another layer only. But later I will try to run first
